### DISCUSSION TITLE
Enhance SSO Capabilities with General OAuth2 Integration

### DISCUSSION THREAD

**Discussion by Spark61** [coollabsio/coolify#2040] [Feature Requests]
I propose to extend the existing single sign-on (SSO) capabilities of Coolify by implementing a general OAuth2 integration framework. This enhancement would enable the integration of a broader range of identity providers such as Keycloak and Authentik, offering more flexibility and security in user authentication.

Benefits:
- Increases the versatility and accessibility of Coolify by supporting a broader array of authentication options.
- Enhances security and user management capabilities.
- Improves user satisfaction by providing more choices for secure access.

I look forward to the community's input and assistance in refining and implementing this feature.

**Comment by LunaVoid** [comment]
Seconding this

**Comment by bmcgonag** [comment]
Also would like this feature added. And, seriously, this is amazing open source software, so thank you for this. I can't imagine how much it must take to build something so incredibly awesome!

**Comment by Mase3206** [comment]
Even if this isn't possible, allow manual user creation and fix the OAuth flow. OAuth users should not have to set a password.

**Comment by ByteAfterlife** [comment]
I hope they add this, especially the keycloak one!

**Reply by pongraczi** [reply]
+1 Univention use keycloak and would be really useful to have this supported.

Don't want to use the provided services, I want to use the self-hosted version.

**Comment by rafaelsilverioit** [comment]
I commented on this existing PR https://github.com/coollabsio/coolify/pull/6696 that is introducing a generic OIDC provider, asking them to get it in for the next release and use it as a starting point to tackle the proposal raised in this discussion.

And as I've written over there, I'll echo here: 

> It's really frustrating to have only a set of cherry picked providers and not be able to use whatever fits our needs, especially for a pretty standardized protocol like OAuth2/OIDC.

**Comment by andrasbacsai** [comment]
Just pasting here [one of the comment](https://github.com/coollabsio/coolify/pull/6696#issuecomment-3614235226) from the PR.

**Comment by NickBouwhuis** [comment]
I would go the OIDC route. This enables providers like Keycloak, Authentik, Zitadel, PocketID and more by implementing a simple standard. Users would only have to enter their discovery URL, client id and secret (if they're not using PKCE).

**Comment by aaronlippold** [comment]
I have been using betterauth on a lot of apps of late, but that is ts/js ... not sure if there is something similar in this ecosystem

**Comment by 2TAP2B** [comment]
any updates here to implement plain oidc? so i can connect my pocket id instance?

**Comment by andrasbacsai** [comment]
It will be released in v4.4
